### PR TITLE
Remove references of Mpin.h

### PR DESCRIPTION
**Pull request by avlaev**

### CODE REVIEW

**Comment by tatiNo5** [comment]
It gives me warnings on building

**Comment by tatiNo5** [comment]
Could you also change the name of the target and the project to be MfaSDK instead of MPinSDK?

**Comment by tatiNo5** [comment]
MpinCoreLib.framework could stay as a name as we're going to remove it sooner or later :)

**Comment by tatiNo5** [comment]
Remove the `iOS SDK API for Milagro (MPin class)` section form the readme too

**Comment by avlaev** [comment]
The project is renamed as well as some of the directories. Warnings are removed(though this is also part of another PR). README has been updated as well.

**Comment by avlaev** [comment]
I removed several files which are autogenerated and can be ignored from git.

**Comment by tatiNo5** [comment]
Please, remove the last line of the readme as the link is not open any more

File: README.md [hunk 1]
```diff
@@ -10,251 +11,12 @@
 ### Building the MFA Mobile SDK
 
 1. Navigate to `<mfa-client-sdk-ios>`
-2. Open `MPinSDK/MPinSDK.xcodeproj`
+2. Open `MfaSDK/MfaSDK.xcodeproj`
 3. Select *Product->Build* from the Xcode menu.
 
 For further details, see [MIRACL MFA Mobile SDK for iOS Documentation](https://devdocs.trust.miracl.cloud/#/mobile-sdk-instructions/)
```
Comment (tatiNo5) [inline README.md]: the link is broken

File: README.md [hunk 2]
```diff
@@ -10,251 +11,12 @@
...
-
-##### `(NSString*) GetClientParam: (const NSString*) key;`
-This method returns the value for a _Client Setting_ with the given key.
-The value is returned as a string always, i.e. when a numeric or a boolean value is expected, the conversion should be handled by the application. 
-Client settings that might interest the applications are:
-* `accessNumberDigits` - The number of Access Number digits that should be entered by the user, prior to calling `FinishAuthenticationAN`.
-* `setDeviceName` - Indicator (`true/false`) whether the application should ask the user to insert a _Device Name_ and pass it to the `MakeNewUser` method.
-* `appID` - The _App ID_ used by the backend. The App ID is a unique ID assigned to each customer or application. It is a hex-encoded long numeric value. The App ID can be used only for information purposes and it does not affect the application's behavior in any way.
 
 ## iOS SDK API for MIRACL MFA (`MPinMFA`)
```
Comment (tatiNo5) [inline README.md]: Please remove the line `It massively resembles the _Apache Milagro_ flavor, while incorporating some functionality is specific to the MIRACL Platform.` and update the next sentence: `Similarly to `MPin`, the `MPinMFA` needs to be initialized.` to be only `The `MPinMFA` object needs to be initialized first.`

File: README.md [hunk 3]
```diff
@@ -301,7 +63,7 @@ Identical and analogical to `MPin`'s [`SetBackend`](#mpinstatus-setbackend-const
 ##### `(id<IUser>) MakeNewUser: (const NSString*) identity deviceName: (const NSString*) devName;`
```
Comment (tatiNo5) [inline README.md]: Please, copy the description of the methods from the MPIN section you've deleted. Now, the links are wrong and the description is not descriptive

**Comment by avlaev** [comment]
I added the changes. I replaced the descriptions for only those methods which have the same signatures as the ones in the removed MPin class. For the others I only removed the broken links.

File: README.md [hunk 4]
```diff
@@ -10,46 +11,21 @@
...
-    CLIENT_SECRET_EXPIRED    // Remote error - re-registration required because server master secret expired
-};
 
-@interface MpinStatus : NSObject
+## iOS SDK API for MIRACL MFA (`MPinMFA`)
 
-@property (nonatomic, readwrite) MPinStatus status;
-@property (nonatomic, strong) NSString* errorMessage;
-@property (NS_NONATOMIC_IOSONLY, getter=getStatusCodeAsString, readonly, copy) NSString* statusCodeAsString;
+This flavor of the SDK should be used to build apps that authenticate users against the _MIRACL MFA Platform_.
```
Comment (tatiNo5) [inline README.md]: Can `MIRACL MFA Platform.` be a link to the platform... trust.miracl.cloud

File: README.md [hunk 5]
```diff
@@ -10,46 +11,21 @@
...
+The `MPinMFA` object needs to be initialized first.
+Most of the methods return a `MpinStatus` object, which is identical to the one used by `MPin`.
 
-@end
-```
+The methods that return `MpinStatus`, will always return status `OK` if successful.
```
Comment (tatiNo5) [inline README.md]: You could mix this sentence with the previous one: `Most of the methods return a `MpinStatus` object as status `OK` means it passed successfully.`

File: README.md [hunk 6]
```diff
@@ -122,10 +101,16 @@ typedef NS_ENUM(NSInteger, UserState)
```
Comment (tatiNo5) [inline README.md]: MakeNewUser description mention Milagro authentication... it should be Miracl authentication

File: README.md [hunk 7]
```diff
@@ -388,33 +192,35 @@ The Platform will also start a user identity verification procedure, by sending
...
+During the call to `ConfirmRegistration` the SDK will make an attempt to retrieve _Client Key_ for the user.
+This attempt will succeed if the user has already been verified/activated but will fail otherwise.
+The method will return status `OK` if the Client Key has been successfully retrieved and `IDENTITY_NOT_VERIFIED` if the identity has not been verified yet.
+If the method has succeeded, the application is expected to get the desired PIN/secret from the end-user and then call `FinishRegistration`, and provide the PIN.
+
+**Note** Using the optional parameter `pushNotificationIdentifier`, the application can provide a platform specific identifier for sending _Push Messages_ to the device. Such push messages might be utilized as an alternative to the _Access Number/Code_, as part of the authentication flow.
 
 ##### `(MpinStatus*) FinishRegistration: (const id<IUser>) user pin0: (NSString*) pin0 pin1: (NSString*) pin1;`
-This method is generally identical and analogical to `MPin`'s [`FinishRegistration`](#mpinstatus-finishregistration-const-idiuser-user-pin-nsstring-pin),
+This method is generally identical and analogical to `MPin`'s `FinishRegistration`,
```
Comment (tatiNo5) [inline README.md]: This description reference to the old mpin method... Correct it

File: README.md [hunk 8]
```diff
@@ -388,33 +192,35 @@ The Platform will also start a user identity verification procedure, by sending
...
 but it allows passing an additional authentication factor as `pin1`. If not needed, `pin1` should be `nil`.
 
 ##### `(MpinStatus*) StartAuthentication: (const id<IUser>) user accessCode: (NSString*) accessCode;`
-This method starts the authentication process for a given `user`.
 It attempts to retrieve the _Time Permits_ for the user, and if successful, will return Status `OK`.
 If they cannot be retrieved, the method will return Status `REVOKED`.
-If this method is successfully completed, the app should read the PIN/secret from the end-user and call one of the `FinishAuthentication` variants to authenticate the user.
```
Comment (tatiNo5) [inline README.md]: Why these two sentences are removed? They are essential part of the description

File: README.md [hunk 9]
```diff
@@ -434,7 +240,7 @@ This method authenticates the end-user for logging into a Web App in a browser s
 The `user` to be authenticated is passed as a parameter, along with his/her secret (`pin0`).
 The `accessCode` associates the authentication with the browser session from which it was obtained.
 
-It is generally identical and analogical to `MPin`'s [`FinishAuthenticationAN`](#mpinstatus-finishauthenticationan-idiuser-user-pin-nsstring-pin-accessnumber-nsstring-an),
+It is generally identical and analogical to `MPin`'s `FinishAuthenticationAN`,
```
Comment (tatiNo5) [inline README.md]: It should be MPinMFA, not MPin

**Comment by avlaev** [comment]
The changes have been added as well as several other changes which are not listed.

File: README.md (hunk 8 above)
Comment (tatiNo5) [reply README.md]: The first sentence also needs to be returned back

File: README.md [hunk 10]
```diff
@@ -532,7 +329,7 @@ The document (`strDoc`) is any free form text that needs to be signed.
 Typically, the customer application will generate a `hash` value for the document that needs to be signed, and will send it to the mobile client app.
 The client app can then verify the correctness of the hash value using this method.
 
-##### `(MpinStatus*) Sign: (id<IUser>) user documentHash: (NSData*) documentHash pin0: (NSString*) pin0 pin1: (NSString*) pin1 epochTime: (double) epochTime authZToken: (NSString*) authZToken result: (BridgeSignature**) result;`
```
Comment (tatiNo5) [inline README.md]: The authZToken description below need to be removed too

**Pull request merged**